sphinx_needs/roles/need_part.py

```diff
@@ -3,31 +3,72 @@
 ---------------
 Provides the ability to mark specific parts of a need with an own id.
 
-Most voodoo is done in need.py
-
 """
 
 from __future__ import annotations
 
 import hashlib
 import re
 from collections.abc import Iterable
-from typing import cast
 
 from docutils import nodes
 from sphinx.application import Sphinx
 from sphinx.environment import BuildEnvironment
+from sphinx.util.docutils import SphinxRole
 from sphinx.util.nodes import make_refnode
 
 from sphinx_needs.data import NeedsInfoType, NeedsPartType
 from sphinx_needs.logging import get_logger, log_warning
 from sphinx_needs.nodes import Need
 
-log = get_logger(__name__)
+LOGGER = get_logger(__name__)
 
 
 class NeedPart(nodes.Inline, nodes.Element):
-    pass
+    @property
+    def title(self) -> str:
+        """Return the title of the part."""
+        return self.attributes["title"]  # type: ignore[no-any-return]
+
+    @property
+    def part_id(self) -> str:
+        """Return the ID of the part."""
+        return self.attributes["part_id"]  # type: ignore[no-any-return]
+
+    @property
+    def need_id(self) -> str | None:
+        """Return the ID of the need this part belongs to."""
+        return self.attributes["need_id"]  # type: ignore[no-any-return]
+
+    @need_id.setter
+    def need_id(self, value: str) -> None:
+        """Set the ID of the need this part belongs to."""
+        self.attributes["need_id"] = value
+
+
+_PART_PATTERN = re.compile(r"\(([\w-]+)\)(.*)", re.DOTALL)
```

Comment:
[re.DOTALL](https://docs.python.org/3/library/re.html#re.DOTALL) matches all symbols and newlines. AFAIK roles cannot span across newlines, correct?

Reply:
roles can span across new lines

Reply:
good to know, have never seen one in the wild